src/v/coding-style.md

```diff
@@ -6,12 +6,50 @@ The document covers style and convention not enforced by clang-format.
 
 Header files have the `.h` extension, source files use the `.cc` extension.
 
+Public header files should go in an `includes` directory within the module they belong to. 
+For example, a module named `foo` should have public header files within `src/v/foo/include/foo`.
+
 Test files go in the `tests` directory of the module they belong to. That is,
 the tests for a feature named `foo` and located in `src/v/foo` will be located
 in `src/v/foo/tests`.
 
 Use `snake_case` naming convention by default for files and language identifiers.
 
+## Documentation
+
+### Modules
+
+Modules themselves should have a `README.md` file explaining the purpose of the module,
```

Comment:
What is a module here? A module is a top-level directory in `src/v`?

Reply:
Any CMake library target that isn't a test - generally this is one to one with a directory, but increasingly these are not all top-level directories.

Open to better naming because of the confusion with C++ modules.

Reply:
> Open to better naming because of the confusion with C++ modules.

I think module is fine but we should probably have one sentence briefly introducing what a module is, to us, before the point we start using that term.